code/code/misc/monster.cc

```diff
@@ -513,44 +513,37 @@ int TMonster::getMobDamage() const
   return max(1, idamrnd);
 }
 
-int TMonster::lookForEngaged(const TBeing *ch)
-{
-  // here's some AI
-  // tank and healer fight mob, tank flees, mob should now beat on
-  // healer (regardless of whether he engaged or not?)
+int TMonster::lookForEngaged() {
   if (isPc())
-    return FALSE;
-  // if someone is attacking me directly, bypass
+    return false;
```

Comment:
I was following the convention used throughout the codebase - tons of functions rely on implicit boolean conversions to int when the function could possibly return a DELETE_X type value. I've never been clearly exactly why they decided to do it that way. `doFlee` itself does the same.

How would you prefer I handle this?

Reply:
Perhaps... perhaps an enum class would be more appropriate? It'd also give the compiler opportunity to verify calling code covers the cases.

Reply:
That seems like a much better way to do it. It would be a big project, as we'd be refactoring a lot of code. I opened https://github.com/sneezymud/sneezymud/issues/396 to track this, as it should definitely be its own project, and I'll resolve this conversation here.